src/HarmonicBarrier.h

```diff
@@ -64,14 +67,42 @@ namespace azplugins
  *          will be raised the first time it is requested.
  *
  */
-class PYBIND11_EXPORT ImplicitEvaporator : public ForceCompute
+class PYBIND11_EXPORT HarmonicBarrier : public ForceCompute
     {
     public:
     //! Constructor
-    ImplicitEvaporator(std::shared_ptr<SystemDefinition> sysdef, std::shared_ptr<Variant> interf);
+    HarmonicBarrier(std::shared_ptr<SystemDefinition> sysdef, std::shared_ptr<Variant> interf);
 
     //! Destructor
-    virtual ~ImplicitEvaporator();
+    virtual ~HarmonicBarrier();
+
+    struct param_type
+        {
+        Scalar k;
+        Scalar offset;
+        Scalar g;
+        Scalar cutoff;
+
+        param_type() : k(0), offset(0), g(0), cutoff(0) { }
+
+        param_type(pybind11::dict params)
+            {
+            k = pybind11::cast<Scalar>(params["k"]);
+            offset = pybind11::cast<Scalar>(params["offset"]);
+            g = pybind11::cast<Scalar>(params["g"]);
+            cutoff = pybind11::cast<Scalar>(params["cutoff"]);
+            }
+
+        pybind11::dict toPython()
+            {
+            pybind11::dict d;
+            d["k"] = pybind11::cast(k);
+            d["offset"] = pybind11::cast(offset);
+            d["g"] = pybind11::cast(g);
+            d["cutoff"] = pybind11::cast(cutoff);
+            return d;
+            }
+        } __attribute__((aligned(16)));
```

Comment:
After you remove `g` and `cutoff`, this struct will be the same as a `Scalar2`. That means that it should be aligned to 8 bytes for 32-bit floats and 16 bytes for 64-bit floats. You can do this like in the LJ potential:
https://github.com/glotzerlab/hoomd-blue/blob/f0415ec516c84f203d99d96f226759034683221a/hoomd/md/EvaluatorPairLJ.h#L151-L155